modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -243,7 +243,109 @@ private void intercept(List<Interceptor> interceptors, Consumer<Interceptor> clo
         });
     }
 
-    private Throwable handleException(Exception e)
+    private class OperationRetryCallBack implements RetryCallback
+    {
+
+        private OperationExecutor operationExecutor;
+        private final OperationContext context;
+        private final List<Interceptor> interceptorList;
+        private OperationExecutionResult operationExecutionResult;
+
+        private OperationRetryCallBack(OperationExecutor operationExecutor, OperationContext context, List<Interceptor> interceptorList)
+        {
+
+            this.operationExecutor = operationExecutor;
+            this.context = context;
+            this.interceptorList = interceptorList;
+        }
+
+        @Override
+        public void doWork(RetryContext retryContext) throws Exception
+        {
+            operationExecutionResult = executeWithInterceptors(operationExecutor, context, interceptorList, new ValueHolder<>());
+
+            if (!operationExecutionResult.isOk())
+            {
+                if (operationExecutionResult.getException() instanceof ConnectionException || (operationExecutionResult.getRetryRequest().isPresent() && operationExecutionResult.getRetryRequest().get().isRetryRequested()))
+                {
+                    throw (Exception) operationExecutionResult.getException();
```

Comment:
By now that point can only reach `Exception`, but nobody knows in a future.
Which would be the nicest (and Muley) way to wrap that `Throwable` into an `Exception`?, just wrapping it in a generic `Exception` or already exist a exception type for this kind of cases?